apps/web/src/api/hooks/use-subscribers.ts

```diff
@@ -0,0 +1,19 @@
+import { useQuery } from 'react-query';
+import { getSubscribersList } from '../subscribers';
+import { useEnvController } from '../../store/use-env-controller';
+import { ISubscriber } from '@novu/shared';
+
+export function useSubscribers(page = 0, limit = 10) {
+  const { environment } = useEnvController();
+  const { data, isLoading } = useQuery<{ data: ISubscriber[]; totalCount: number; pageSize: number }>(
+    ['subscribersList', environment?._id],
```

Comment:
the query key should include the `page, limit` variables, otherwise, it won't trigger the API request if the page/limit changes... so in the result, the `page` in the pagination will change but the table data will be the same...

Reply:
https://user-images.githubusercontent.com/2607232/188314229-c475a5c0-56d2-4e45-b4e2-3f25f0f9e8fc.mov

Reply:
fixed!